src/inspector_agent.cc

```diff
@@ -592,6 +613,21 @@ class NodeInspectorClient : public V8InspectorClient {
     return env_->isolate_data()->platform()->CurrentClockTimeMillis();
   }
 
+  std::unique_ptr<StringBuffer> resourceNameToUrl(
+      const StringView& resource_name_view) override {
+    static const char FILE_URL_PREFIX[] = "file://";
+
+    std::string resource_name =
+        protocol::StringUtil::StringViewToUtf8(resource_name_view);
+    if (!IsFilePath(resource_name))
+      return nullptr;
+    for (size_t i = 0; i < resource_name.length(); ++i) {
+      if (resource_name[i] == '\\')
+        resource_name[i] = '/';
+    }
+    return Utf8ToStringView(FILE_URL_PREFIX + resource_name);
```

Comment:
`FILE_URL_PREFIX + resource_name` is not sufficient for a full file URL conversion. You would need a C++ version of something like

https://github.com/nodejs/node/blob/503fd55a35ce056d636c15811c07b51126af271a/lib/internal/url.js#L1408-L1415

Could you add that to src/node\_url.cc and use it from here?

Reply:
done